client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -336,14 +362,14 @@ public void processingFailure(Exception error) {
             state.failConnection(error);
         }
     }
-    
+
     /**
-     * @see SegmentOutputStream#write(java.nio.ByteBuffer,
-     *      java.util.concurrent.CompletableFuture)
+     * @see SegmentOutputStream#write(PendingEvent)
      */
     @Override
     @Synchronized
     public void write(PendingEvent event) throws SegmentSealedException {
```

Comment:
seal() and write() are never called concurrently since the segmentSealedCallback and write are mutually exclusive. Also, annotating `seal()` with synchronized will cause `seal()` to be blocked when a SegmentSealed is encountered during `flush()`